benchmark/worker/benchmark_client.go

```diff
@@ -0,0 +1,330 @@
+		}
+	}
+
+	grpclog.Printf(" - rpcs per chann: %v", setup.OutstandingRpcsPerChannel)
+	grpclog.Printf(" - channel number: %v", setup.ClientChannels)
+
+	rpcCountPerConn, connCount := int(setup.OutstandingRpcsPerChannel), int(setup.ClientChannels)
+
+	grpclog.Printf(" - load params: %v", setup.LoadParams)
+	var dist *int
+	switch lp := setup.LoadParams.Load.(type) {
+	case *testpb.LoadParams_ClosedLoop:
+	case *testpb.LoadParams_Poisson:
+		grpclog.Printf("   - %v", lp.Poisson)
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", setup.LoadParams)
+		// TODO poisson
+	case *testpb.LoadParams_Uniform:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", setup.LoadParams)
+	case *testpb.LoadParams_Determ:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", setup.LoadParams)
+	case *testpb.LoadParams_Pareto:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", setup.LoadParams)
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "unknown load params: %v", setup.LoadParams)
+	}
+
+	grpclog.Printf(" - rpc type: %v", setup.RpcType)
+	var rpcType string
+	switch setup.RpcType {
+	case testpb.RpcType_UNARY:
+		rpcType = "unary"
+	case testpb.RpcType_STREAMING:
+		rpcType = "streaming"
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "unknown rpc type: %v", setup.RpcType)
+	}
+
+	grpclog.Printf(" - histogram params: %v", setup.HistogramParams)
+	grpclog.Printf(" - server targets: %v", setup.ServerTargets)
+
+	conns := make([]*grpc.ClientConn, connCount)
+
+	for connIndex := 0; connIndex < connCount; connIndex++ {
+		conns[connIndex] = benchmark.NewClientConn(setup.ServerTargets[connIndex%len(setup.ServerTargets)], opts...)
+	}
+
+	bc := benchmarkClient{
+		histogram: stats.NewHistogram(stats.HistogramOptions{
+			NumBuckets:     int(math.Log(setup.HistogramParams.MaxPossible)/math.Log(1+setup.HistogramParams.Resolution)) + 1,
+			GrowthFactor:   setup.HistogramParams.Resolution,
+			BaseBucketSize: (1 + setup.HistogramParams.Resolution),
+			MinValue:       0,
+		}),
+		stop:          make(chan bool),
+		lastResetTime: time.Now(),
+	}
+
+	switch rpcType {
+	case "unary":
+		if dist == nil {
+			bc.doCloseLoopUnaryBenchmark(conns, rpcCountPerConn, payloadReqSize, payloadRespSize)
+		}
+		// TODO else do open loop
+	case "streaming":
+		if dist == nil {
+			bc.doCloseLoopStreamingBenchmark(conns, rpcCountPerConn, payloadReqSize, payloadRespSize, payloadType)
+		}
+		// TODO else do open loop
+	}
+
+	return &bc, nil
+}
+
+func (bc *benchmarkClient) doCloseLoopUnaryBenchmark(conns []*grpc.ClientConn, rpcCountPerConn int, reqSize int, respSize int) {
+	clients := make([]testpb.BenchmarkServiceClient, len(conns))
+	for ic, conn := range conns {
+		clients[ic] = testpb.NewBenchmarkServiceClient(conn)
+		// Do some warm up.
+		for j := 0; j < 10; j++ {
+			benchmark.DoUnaryCall(clients[ic], 1, 1)
+		}
+	}
```

Comment:
move the warmup into the iteration in the following. The only difference is that you do not insert warmup results into the histogram.